### PR TITLE
Fix rendering link attachment preview with other attachment types

### PR DESCRIPTION
**Pull request by nuno-vieira**
### 🔗 Issue Link
Fixes IOS-545

### 🎯 Goal

Fix rendering link attachment preview with other attachment types

### 🛠 Implementation

I've added a new condition, "hasOnlyLinks", to render the link attachment. This works, but I feel like the best solution would be for this to be the responsibility of `MessageTypeResolving`. The problem is that the name of the method is `hasLinkAttachment()`, so if I added this boolean inside, the name loses its true meaning because we are checking instead "onlyHasLinkAttachment()`. 

So it feels like this `MessageTypeResolving` would probably be better for the names to be something like: `shouldRenderLinkAttachment()`. This way, additional logic could be added to these methods, and it would be easier for customers to provide their own logic whether they want an attachment to be rendered or not. Maybe we can deprecate the old names, and introduce the "shouldRender" prefix instead, but probably out of scope for this PR.

### 🧪 Testing

Add an image and a link in the same message. It should only render the image preview, not the link preview.

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created)

### CODE REVIEW

File: Sources/StreamChatSwiftUI/ChatChannel/MessageList/MessageView.swift [hunk 1]
```diff
@@ -44,7 +44,8 @@ public struct MessageView<Factory: ViewFactory>: View {
             } else if let poll = message.poll {
                 factory.makePollView(message: message, poll: poll, isFirst: isFirst)
             } else if !message.attachmentCounts.isEmpty {
-                if messageTypeResolver.hasLinkAttachment(message: message) {
+                let hasOnlyLinks = message.attachmentCounts.keys.allSatisfy { $0 == .linkPreview }
```
[the remark this answers is not in the document]
Comment (nuno-vieira) [reply Sources/StreamChatSwiftUI/ChatChannel/MessageList/MessageView.swift:47]: The thing is that we are doing this kinda logic with video attachment as well, where we only render the video attachment view if there is not image attachment as well:
```
messageTypeResolver.hasVideoAttachment(message: message)
                    && !messageTypeResolver.hasImageAttachment(message: message)
```
That is why opted for this approach as well 🤔

[the remark this answers is not in the document]
Comment (martinmitrevski) [reply Sources/StreamChatSwiftUI/ChatChannel/MessageList/MessageView.swift:47]: I see. Maybe we can just put the check directly after the &&, so it's executed only when `hasLinkAttachment` is true?

[the remark this answers is not in the document]
Comment (nuno-vieira) [reply Sources/StreamChatSwiftUI/ChatChannel/MessageList/MessageView.swift:47]: Yes makes sense 👍

[the remark this answers is not in the document]
Comment (nuno-vieira) [reply Sources/StreamChatSwiftUI/ChatChannel/MessageList/MessageView.swift]: Done ✅

**Review by martinmitrevski — APPROVED** [review]

**Pull request merged**